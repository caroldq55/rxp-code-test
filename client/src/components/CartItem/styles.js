import styled from "styled-components";

const CartItemContainer = styled.div`
  margin: 5px;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const ItemName = styled.span`
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
`;

export { CartItemContainer, ItemName };
