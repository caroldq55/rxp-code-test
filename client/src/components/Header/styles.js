import styled from 'styled-components';

const HeaderContainer = styled.div`
  background-color: #f4f4f4;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1;
`;

const Brand = styled.div`
  padding: 5px 20px;
  font-weight: bold;
  font-size: 25px;
`;

export { HeaderContainer, Brand };
