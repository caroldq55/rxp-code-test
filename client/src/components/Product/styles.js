import styled from 'styled-components';

const ProductContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 20px;
  margin: 5px;
  position: relative;
`;

const ProductImage = styled.img`
  width: 180px;
`;

const ProductInfo = styled.div`
  margin-top: 10px;
  font-weight: 500;
`;

const ProductName = styled.div`
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  cursor: pointer;
`;

const ProductPrice = styled.div`
  font-weight: bold;
  color: #333;
`;

const ButtonWrapper = styled.div`
  text-align: center;
  margin-top: 5px;
`;

export {
  ProductContainer,
  ProductImage,
  ProductInfo,
  ProductPrice,
  ProductName,
  ButtonWrapper
};
