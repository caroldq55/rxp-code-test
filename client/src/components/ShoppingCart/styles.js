import styled from 'styled-components';

const ShopingCartContainer = styled.div`
  margin: 5px;
  border: 0.0625em solid #dcdfe1;
  border-radius: 0.25em;
  min-height: 50px;
`;

const Heading = styled.div`
  padding: 5px 10px;
  background-color: #ffd633;
  font-weight: bold;
`;

const Content = styled.div`
  padding: 10px;
`;

const TotalPrice = styled.div`
  font-weight: bold;
  margin-top: 10px;
`;

export { ShopingCartContainer, Heading, Content, TotalPrice };
