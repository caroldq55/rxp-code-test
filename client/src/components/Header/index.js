import React from 'react';
import PropTypes from 'prop-types';

import { HeaderContainer, Brand } from './styles';

const Header = props => (
  <HeaderContainer>
    <Brand>{props.title}</Brand>
  </HeaderContainer>
);

export default Header;

Header.propTypes = {
  title: PropTypes.string.isRequired
};
