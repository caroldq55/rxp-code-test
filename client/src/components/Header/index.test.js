import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Header from '.';
import { Brand } from './styles';

Brand.displayName = 'Brand';

describe('<Header/>', () => {
  it('should render Header correctly', () => {
    const wrapper = shallow(<Header title='Telstra' />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Brand').contains('Telstra')).toBe(true);
  });
});
