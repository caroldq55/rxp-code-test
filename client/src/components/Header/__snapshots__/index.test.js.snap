// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should render Header correctly 1`] = `ShallowWrapper {}`;
